lib/unleash/backup.ex

```diff
@@ -0,0 +1,57 @@
+defmodule Unleash.Backup do
+  @moduledoc """
+  This module handles reading and writing to a backup file. Used as a backup
+  data store in case we can't fetch data from unleash or our cache
+
+  The following configuration values are used:
+
+  Config.backup_file(): path to which our feature state gets saved
+  """
+
+  require Logger
+
+  alias Unleash.Config
+  alias Unleash.Feature
+
+  @derive Jason.Encoder
+  defstruct version: "1", features: []
+
+  def read_state(backup_file \\ Config.backup_file()) do
+    if File.exists?(backup_file) do
+      backup_file
+      |> File.read!()
+      |> Jason.decode!()
+      |> from_map()
+    else
+      []
+    end
+  end
+
+  def write_state(
+        features,
+        backup_dir \\ Config.backup_dir(),
+        backup_file \\ Config.backup_file()
+      ) do
+    if not File.dir?(backup_dir) do
+      backup_dir
+      |> File.mkdir_p!()
+    end
```

Comment:
The [non-bang version](https://hexdocs.pm/elixir/File.html#mkdir_p/1) of this function might be simpler to use as it follows the Unix approach to `mkdir -p`: blindly create the entire directory structure.
```suggestion
    :ok = File.mkdir_p((backup_dir)
```